pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -186,14 +189,24 @@ func RunBuildChain(f *clientcmd.Factory, cmd *cobra.Command, args []string) erro
 		// Set the specified repo as the only one to output dependencies for
 		repos[repo] = tags
 	} else {
-		// Get all image repositories from build configurations
-		repos = getRepos(buildConfigList)
-
-		// Make sure from now on that the --all flag is true
-		// since we are building dependency trees for every
-		// image repository available either in the current
-		// namespace or in every namespace
-		all = true
+		// all the tags of all the imagestreams across all the namespaces
+		for _, ns := range namespaces {
```

Comment:
@smarterclayton @kargakis i went ahead and reworked this to not make an imagestream call per project, but as noted earlier, we are still (and always have been) making a buildconfig call per namespace because the API will not return "all buildconfigs for all projects I can view" in a single call.